src/utils/NucleicAcidSearchEngine.cpp

```diff
@@ -879,13 +899,20 @@ class NucleicAcidSearchEngine :
       // @TODO: use charge-specific RTs?
       vector<Int> charges;
       vector<double> rts;
-      for (const auto& pair : entry.second)
+      for (const auto& charge_pair : entry.second)
       {
-        charges.push_back(pair.first);
-        rts.insert(rts.end(), pair.second.begin(), pair.second.end());
+        charges.push_back(charge_pair.first);
+        // use intensity-weighted mean of precursor RTs as "apex" RT:
+        double weighted_rt = 0.0, total_weight = 0.0;
+        for (const auto& rt_pair : charge_pair.second)
+        {
+          weighted_rt += rt_pair.first * rt_pair.second;
+          total_weight += rt_pair.first;
+        }
+        rts.push_back(weighted_rt / total_weight);
       }
       tsv << name << ef << 0 << ListUtils::concatenate(charges, ",");
-      // @TODO: do something more sophisticated for the RT?
+      // overall target RT is median over all charge states:
```

Comment:
without seeing the full picture this sounds a bit dangerous

Reply:
My reasoning is that LC happens before ionisation, so in principle all charge states must have the same RT. I'm trying to alleviate problems from patchy MS2 sampling by combining data from multiple charge states.

Reply:
I don’t know much about that but I thought that this Depends on the adduct.

Reply:
Not sure if that's what you mean, but the RT calculations are done separately for every combination of oligonucleotide and adduct. So you can get different RTs for the same oligo with different adducts.

Reply:
so every oligo + adduct pair get 1 target RT. If several charge states of the same oligo + adduct pair exist then also the median of all RTs is returned?

Reply:
Yes. One final RT per oligo/adduct pair, which is calculated from all charge states of that oligo/adduct pair.

Reply:
Ok fine with me. I only see problems with e.g. deuterium as charge carrier but this sounds kind of artificial.